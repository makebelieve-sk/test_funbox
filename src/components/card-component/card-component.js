import React, { useState } from 'react';

export const CardComponent = ({ product, selected, buyMode, i, text, mouseEnter, mouseLeave, checked, change, disabled }) => {
    const [title, setTitle] = useState(null);

    return (
        <div 
            className="product-wrapper" 
            key={product.id} 
            onMouseEnter={mouseEnter ? mouseEnter.bind(this, i, setTitle) : null}
            onMouseLeave={mouseLeave ? mouseLeave.bind(this, i, setTitle) : null}
        >
            <label 
                className="choose_check" 
                htmlFor={`choose-${product.id}`} 
            >
                <input 
                    checked={ checked ? selected[i] : null}
                    className={`choose_input ${buyMode} ${disabled}`} 
                    type="checkbox" 
                    name="choose" 
                    id={`choose-${product.id}`}
                    onChange={change ? change.bind(this, i) : null}
                />
                <span />

                <div className="block_in_card">
                    <p className={`product_title ${disabled}`}>{ title ? title : product.title }</p>
                    <p className={`product_name ${disabled}`}>{ product.name }</p>
                    <p className={`product_with ${disabled}`}>{ product.with }</p>
                    <p className={`product_description ${disabled}`}>{ product.description_line1 }</p>
                    <p className={`product_description ${disabled}`}>{ product.description_line2 }</p>
                    <p className={`product_description ${disabled}`}>{ product.description_line3 }</p>
                </div>

                <div className="product_weight">
                    <p className="product_weight_number">{ product.weight }</p>
                    <p className="product_weight_text">кг</p>
                </div>

                { <p className={`product_text ${disabled}_text`}>{ text }</p> }
            </label>
        </div>
    )
};