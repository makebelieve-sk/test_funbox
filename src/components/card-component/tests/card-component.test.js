import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { CardComponent } from '../card-component';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2, prop3, prop4, prop5, prop6, prop7, prop8, prop9, prop10) => shallow(
    <CardComponent product={prop1} selected={prop2} buyMode={prop3} i={prop4} text={prop5} mouseEnter={prop6} mouseLeave={prop7} checked={prop8} change={prop9} disabled={prop10} />
);

const mock = {
    product: {
        id: 0,
        title: `Сказочное заморское яство`,
        hoverTitle: `Котэ не одобряет?`,
        name: `Нямушка`,
        with: `с фуа-гра`,
        description_line1: `10 порций`,
        description_line2: `мышь в подарок`,
        weight: `0,5`,
        quantity: 1,
        text: `Чего сидишь? Порадуй котэ, `,
        buy: true
    },
    selected: [false, false, false, false],
    buyMode: `string`,
    i: 0,
    text: `some text`,
    mouseEnter: jest.fn(),
    mouseLeave: jest.fn(),
    checked: false,
    change: jest.fn(),
    disabled: `some string`
};

describe(`CardComponent component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.product, mock.selected, mock.buyMode, mock.i, mock.text, mock.mouseEnter, mock.mouseLeave, mock.checked, mock.change, mock.disabled);
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.product-wrapper');
        expect(wrapper.length).toBe(1);
    });

    it(`CardComponent correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<CardComponent 
                product={mock.product}
                selected={mock.selected}
                buyMode={mock.buyMode}
                i={mock.i}
                text={mock.text}
                mouseEnter={mock.mouseEnter}
                mouseLeave={mock.mouseLeave}
                checked={mock.checked}
                change={mock.change}
                disabled={mock.disabled}
            />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});