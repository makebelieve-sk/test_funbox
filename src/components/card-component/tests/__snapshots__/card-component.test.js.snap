// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent component CardComponent correctly renders after relaunch 1`] = `
<div
  className="product-wrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="choose_check"
    htmlFor="choose-0"
  >
    <input
      checked={null}
      className="choose_input string some string"
      id="choose-0"
      name="choose"
      onChange={[Function]}
      type="checkbox"
    />
    <span />
    <div
      className="block_in_card"
    >
      <p
        className="product_title some string"
      >
        Сказочное заморское яство
      </p>
      <p
        className="product_name some string"
      >
        Нямушка
      </p>
      <p
        className="product_with some string"
      >
        с фуа-гра
      </p>
      <p
        className="product_description some string"
      >
        10 порций
      </p>
      <p
        className="product_description some string"
      >
        мышь в подарок
      </p>
      <p
        className="product_description some string"
      />
    </div>
    <div
      className="product_weight"
    >
      <p
        className="product_weight_number"
      >
        0,5
      </p>
      <p
        className="product_weight_text"
      >
        кг
      </p>
    </div>
    <p
      className="product_text some string_text"
    >
      some text
    </p>
  </label>
</div>
`;
